codegen/module_system.go

```diff
@@ -34,7 +34,11 @@ import (
 
 // NewDefaultModuleSystem creates a fresh instance of the default zanzibar
 // module system (clients, endpoints)
-func NewDefaultModuleSystem(h *PackageHelper) (*module.System, error) {
+func NewDefaultModuleSystem(
+	configDirName string,
+	middlewareConfig string,
+	h *PackageHelper,
```

Comment:
I don't feel like it belongs in either place, but the PackageHelper gives us information about the structure of the package. With middlewares in this current mode, they are a permanent fixture of the package, which is arguably why the package helper does make some sense. It also have the benefit that existing consumers initialize the package helper in the current way and this configuration can be defaulted.